src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -430,30 +429,30 @@ internal void StatementCompletedHandler(object sender, StatementCompletedEventAr
         /// </summary>
         /// <param name="sender">Object that fired the event</param>
         /// <param name="args">Arguments from the event</param>
-        private void StoreDbMessage(object sender, SqlInfoMessageEventArgs args)
+        private void ServerMessageHandler(object sender, SqlInfoMessageEventArgs args)
         {
-            resultMessages.Add(new ResultMessage(args.Message));
+            SendMessage(args.Message, false).Wait();
```

Comment:
sendMessage returns null if the message is null, should you check for null here before running .Wait()?